cgen: using a mut method of mut generic array parameter makes undefined symbol error

### Describe the bug

An array of generic type for parameters are frequently used in many cases.
However, directly writing any mutable methods of the mutable parameters makes undefined symbol error.

### Reproduction Steps

```
fn grow[T](mut arr []T) {
        arr.grow_cap(10)
}

mut arr := []int{}
grow(mut arr)
```

### Expected Behavior

compile, and it should use `array_grow_cap`.

### Current Behavior

`Array_int_grow_cap` is used, and it makes an error below.

```
/tmp/v_1000/method.01JJC2RJJMY6ZM27DYT2YEVTJ3.tmp.c:5738: warning: implicit declaration of function 'Array_int_grow_cap'
/home/lemon/git/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
tcc: error: undefined symbol 'Array_int_grow_cap'
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 da5bb68

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.9 7eec8b1.da5bb68
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.1.21.2-microsoft-standard-WSL2+ (root@DESPTOP-LEMON) (gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #2 SMP PREEMPT_DYNAMIC Fri Jul 21 23:54:26 JST 2023 (WSL 2)
|Processor           |16 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11700KF @ 3.60GHz
|Memory              |30.05GB/31.3GB
|                    |
|V executable        |/home/lemon/git/v/v
|V last modified time|2025-01-24 04:32:39
|                    |
|V home dir          |OK, value: /home/lemon/git/v
|VMODULES            |OK, value: /home/lemon/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/lemon/vtest
|                    |
|Git version         |git version 2.48.1
|V git status        |0.3.5-3653-gda5bb68b-dirty (3 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |cc (GCC) 14.2.1 20240910
|gcc version         |gcc (GCC) 14.2.1 20240910
|clang version       |clang version 19.1.6
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.40
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1190,6 +1190,13 @@ fn (mut g Gen) gen_array_method_call(node ast.CallExpr, left_type ast.Type, left
 			}
 			g.write(')')
 		}
+		'grow_cap', 'grow_len' {
+			g.write('array_${node.name}(')
+			g.gen_arg_from_type(left_type, node.left)
+			g.write(', ')
+			g.expr(node.args[0].expr)
+			g.write(')')
+		}
 		'first', 'last', 'pop' {
 			mut noscan := ''
 			array_info := left_sym.info as ast.Array

--- a/vlib/v/tests/builtin_arrays/array_grow_cap_test.v
+++ b/vlib/v/tests/builtin_arrays/array_grow_cap_test.v
@@ -1,0 +1,9 @@
+fn grow[T](mut arr []T) {
+	arr.grow_cap(10)
+	unsafe { arr.grow_len(10) }
+}
+
+fn test_main() {
+	mut arr := []int{}
+	grow(mut arr)
+}

